stdlib/src/collections/dict.mojo

```diff
@@ -454,6 +454,40 @@ struct Dict[K: KeyElement, V: CollectionElement](
         self._index = existing._index.copy(existing._reserved)
         self._entries = existing._entries
 
+    @staticmethod
+    fn fromkeys(keys: List[K], value: V) -> Self:
```

Comment:
seems like a good idea but calling `pop` would inverse the order of the keys, and also calling `pop(0)` could be expensive I guess? since all the elements would need to be moved

Reply:
MMMhhhh indeed. I see two possibilities here to avoid a copy:
1) Short term: Use the same trick that we have in `List.extends()` where we use the pointers to "consume" the objects in order. This makes use of pointers and is pretty unsafe.
2) When we have a https://docs.python.org/3/library/collections.html#collections.deque then we can create a deque by consuming a `List`, and then we can pop at the start with little to no cost.

Reply:
I do not see how the same trick used in `extend` could be applied here, in list al the elements are one after the other in memory but that's not the case for a `Dict` I'm missing something?
Using a `Deque` could be a good option when it exists I guess.

Either way I would say that optimizing this could be done later, if needed, not sure how much this method is going to be used (personally I have rarely seen it used in Python).

Reply:
Indeed, I think it's trickier than I imagined because the elements in the dict are not contiguous, it's hard to work with pointers. I will open an issue to add deque to the stdlib. Deque could also be used to make a safe implementation of `List.extends()`.